src/providers/wfs/oapif/qgsoapifschemarequest.cpp

```diff
@@ -0,0 +1,256 @@
+
+  QTextCodec::ConverterState state;
+  QTextCodec *codec = QTextCodec::codecForName( "UTF-8" );
+  Q_ASSERT( codec );
+
+  const QString utf8Text = codec->toUnicode( buffer.constData(), buffer.size(), &state );
+  if ( state.invalidChars != 0 )
+  {
+    mErrorCode = QgsBaseNetworkRequest::ApplicationLevelError;
+    mErrorMessage = errorMessageWithReason( tr( "Invalid UTF-8 content" ) );
+    return;
+  }
+
+  // Key is the explicit x-ogc-propertySeq or a synthetized one
+  std::map<int, QgsField> fields;
+  int seqNumber = 1;
+  int largestSeqNumber = 1;
+  bool seqNumbersAreOk = true;
+  try
+  {
+    const json j = json::parse( utf8Text.toStdString() );
+
+    if ( j.is_object() && j.contains( "properties" ) )
+    {
+      const json jProperties = j["properties"];
+      if ( jProperties.is_object() )
+      {
+        for ( const auto &[key, val] : jProperties.items() )
+        {
+          if ( val.is_object() && val.contains( "type" ) )
+          {
+            QgsField field( QString::fromStdString( key ), QMetaType::Type::QString );
+
+            const json jType = val["type"];
+            if ( jType.is_string() )
+            {
+              const QString type = QString::fromStdString( jType.get<std::string>() );
+              if ( type == QLatin1String( "integer" ) )
+              {
+                field.setType( QMetaType::Type::LongLong );
+              }
+              else if ( type == QLatin1String( "number" ) )
+              {
+                field.setType( QMetaType::Type::Double );
+              }
+              else if ( type == QLatin1String( "boolean" ) )
+              {
+                field.setType( QMetaType::Type::Bool );
+              }
+              else if ( type == QLatin1String( "object" ) )
+              {
+                field.setType( QMetaType::Type::QVariantMap );
+              }
+              else if ( type == QLatin1String( "array" ) )
+              {
+                field.setType( QMetaType::Type::QVariantList );
+              }
+              else if ( type == QLatin1String( "string" ) )
+              {
+                if ( val.contains( "format" ) )
+                {
+                  const json jFormat = val["format"];
+                  if ( jFormat.is_string() )
+                  {
+                    const QString format = QString::fromStdString( jFormat.get<std::string>() );
+                    if ( format == QLatin1String( "date-time" ) )
+                    {
+                      field.setType( QMetaType::Type::QDateTime );
+                    }
+                    else if ( format == QLatin1String( "date" ) )
+                    {
+                      field.setType( QMetaType::Type::QDate );
+                    }
+                  }
+                }
+              }
+              else
+              {
+                QgsDebugMsgLevel( QStringLiteral( "Unhandled OGC API Features schema type: " ) + type, 4 );
```

Comment:
```suggestion
                QgsDebugMsgLevel( QStringLiteral( "Unhandled OGC API Features schema type: " ) + type, 3 );
```
lower a bit the level?

Reply:
done